src/service.rs

```diff
@@ -1393,7 +1392,8 @@ impl Service {
 
     /// Get the service status from the system.
     pub fn query_status(&self) -> crate::Result<ServiceStatus> {
-        let mut raw_status = unsafe { mem::zeroed::<winsvc::SERVICE_STATUS_PROCESS>() };
+        let mut raw_status =
+            unsafe { mem::MaybeUninit::<winsvc::SERVICE_STATUS_PROCESS>::zeroed().assume_init() };
```

Comment:
The docs for `mem::zeroed()` states that it has the same effect as what you changed this into. So this makes no difference in safety or correctness.

The way we probably want to use `MaybeUninit` is with `MaybeUninit::uninit()` and then getting a pointer to it with `raw_status.as_mut_ptr()` in order to pass it over FFI to be filled in. And then after it has been filled in call `assume_init()` on it. Like how the docs describe it in https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html. The point is to keep the actual C struct inside the `MaybeUninit` until it has been properly initialized, and then take it out with `assume_init()`. If we take it out before it's initialized `MaybeUninit` does not help us much.

Reply:
That makes sense, I'll update the PR accordingly, then